src/utils.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2025 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pandas as pd
+from plotly.colors import sample_colorscale
+import plotly.graph_objs as go
+
+from data import DataSet
+from demo_configs import COLOR_SCALE, GRAPH_FONT_SIZE
+
+
+def draw_bar_chart(hover_data: dict, selected_features: list, data: DataSet, show_red: bool) -> go.Figure:
```

Comment:
If this is can be None like the docstring says then it should probably be typed like this: (you'll need to import `from typing import Optional
`)

```suggestion
def draw_bar_chart(hover_data: dict, selected_features: Optional[list], data: DataSet, show_red: bool) -> go.Figure:
```